src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -17832,22 +17955,311 @@ static RefByteContainer M32([UnscopedRef] ref ByteContainer bc)
+                //             rs1 = rs.ToRSOut(); // 6
+                Diagnostic(ErrorCode.ERR_RefReturnOnlyParameter, "rs").WithArguments("rs").WithLocation(63, 19));
+        }
+
+        [Fact, WorkItem(63526, "https://github.com/dotnet/roslyn/issues/63526")]
+        public void ReturnOnlyScope_02()
+        {
+            var source = """
+                using System.Diagnostics.CodeAnalysis;
+
+                #pragma warning disable 8321 // unused local function
+                static bool condition() => false;
+
+                static void M1(ref S p1, ref S p2) {
+                    p2.field = ref p1.field; // 1
+                    p2.refField = ref p1.refField; // Okay
+                }
+
+                static void M2(ref S p1, out S p2) {
+                    p2 = default;
+                    p2.field = ref p1.field; // 2
+                    p2.refField = ref p1.refField; // Okay
+                }
+
+                static void M3(out S p1, ref S p2) {
+                    p1 = default;
+                    p2.field = ref p1.field; // 3
+                    p2.refField = ref p1.refField; // Okay
+                }
+
+                // The [UnscopedRef] moves `out` to default RSTS which is *return only*
+                static void M4([UnscopedRef] out S p1, ref S p2) {
+                    p1 = default;
+                    p2.field = ref p1.field; // 4
+                    p2.refField = ref p1.refField; // Okay
+                }
+
+                static void M5([UnscopedRef] ref S p1, ref S2 p2) {
+                    p2 = Inner1(ref p1); // 5
+                    p2 = Inner2(ref p1); // Okay
+                    static S2 Inner1([UnscopedRef] ref S s) => new S2 { S = s };
+                    static S2 Inner2(scoped ref S s) => new S2 { S = s };
+                }
+
+                static void M6([UnscopedRef] ref S p1, out S2 p2) {
+                    p2 = Inner1(ref p1); // 6
+                    p2 = Inner2(ref p1); // Okay
+                    static S2 Inner1([UnscopedRef] ref S s) => new S2 { S = s };
+                    static S2 Inner2(scoped ref S s) => new S2 { S = s };
+                }
+
+                static void M7(scoped ref S p1, ref S2 p2) {
+                    p2 = Inner1(ref p1); // 7
+                    p2 = Inner2(ref p1); // Okay
+                    static S2 Inner1([UnscopedRef] ref S s) => new S2 { S = s };
+                    static S2 Inner2(scoped ref S s) => new S2 { S = s };
+                }
+
+                static S2 M8(scoped ref S p) {
+                    if (condition()) return Inner1(ref p); // 8
+                    if (condition()) return Inner2(ref p); // Okay
+
+                    throw null!;
+
+                    static S2 Inner1([UnscopedRef] ref S s) => new S2 { S = s };
+                    static S2 Inner2(scoped ref S s) => new S2 { S = s };
+                }
```

Comment:
Consider adding:
```csharp
static S2 M9([UnscopedRef] ref S p)
{
    if (condition()) return Inner1(ref p);
    return Inner2(ref p); // 9

    static S2 Inner1([UnscopedRef] ref S s) => new S2 { S = s };
    static S2 Inner2(ref S s) => new S2 { S = s };
}
```

Reply:
Did you expect `return Inner2(ref p)` to be an error here? Why? It looks like `ref S p` has `ReturnOnly` scope, so it's OK to return `ref S` through `Inner2` (although that doesn't actually happen, that's what the signature of `Inner2` indicates)

Reply:
> Did you expect return Inner2(ref p) to be an error here? 

Think for this PR where the RSTE of a `ref` to `ref struct` in between we should be

Reply:
my brain is constantly switching the RSTE of RTRS between ReturnOnly and CurrentMethod. we need to get #64122 done so the pain will stop